tests/PySys/monitoring/monitoring_smaller_interval/run.py

```diff
@@ -94,14 +94,17 @@ def validate_json(self):
             if not self.validate_time():
                 reason = "time validation failed in message: " + str(line)
                 self.abort(False, reason)
-            if not self.validate_temperature():
-                reason = "temperature stat validation failed in message: " + \
-                    str(line)
-                self.abort(False, reason)
-            if not self.validate_pressure():
-                reason = "pressure stat validation failed in message: " + \
-                    str(line)
-                self.abort(False, reason)
+            if "temperature" in self.js_msg:
+                if not self.validate_temperature():
+                    reason = "temperature stat validation failed in message: " + \
+                        str(line)
+                    self.abort(False, reason)
+            if "pressure" in self.js_msg:
+                if not self.validate_pressure():
+                    reason = "pressure stat validation failed in message: " + \
+                        str(line)
+                    self.abort(False, reason)
+        return True
```

Comment:
With these changes we need an additional validation that the temperature and pressure are appearing at least once, either in a single JSON message or split across multiple messages. But that "exactly once" logic needs to be tested.